### PR TITLE
feat: typescript on Field Component

### PR DESCRIPTION
**Pull request by cnairiwttj**

### CODE REVIEW

File: packages/Field/index.tsx [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
 
-export const Field = forwardRef(
+export type Size = 'sm' | 'md' | 'lg'
+
+export interface FieldProps {
+  checked?: boolean
+  children?: React.ReactNode
+  component: React.ForwardRefRenderFunction<HTMLInputElement, Omit<FieldProps, 'component'>>
+  connected?: boolean
+  dataTestId?: string
```
Comment (theo-mesnil) [inline packages/Field/index.tsx:20]: it's on WuiProps

File: packages/Field/index.tsx [hunk 2]
```diff
@@ -1,17 +1,44 @@
...
+  size?: Size
+  touched?: boolean
+  type?: string
+  warning?: string
+  required?: boolean
+  ref: React.ForwardedRef<HTMLDivElement>
+  variant?: VariantReturn
+}
+
+export const Field = forwardRef<HTMLInputElement, FieldProps>(
```
Comment (theo-mesnil) [inline packages/Field/index.tsx:41]: missing WuiProps

File: packages/Field/index.tsx [hunk 3]
```diff
@@ -1,17 +1,44 @@
...
+export interface FieldProps {
+  checked?: boolean
+  children?: React.ReactNode
+  component: React.ForwardRefRenderFunction<HTMLInputElement, Omit<FieldProps, 'component'>>
+  connected?: boolean
+  dataTestId?: string
+  disabled?: boolean
+  disabledIcon?: JSX.Element
+  error?: string
+  flexDirection?: flexDirection
```
Comment (theo-mesnil) [inline packages/Field/index.tsx:24]: it's on WuiProps

File: packages/Field/styles.ts [hunk 4]
```diff
@@ -17,7 +19,14 @@ const checkableFieldStyles = css`
...
-export const Field = styled('div').withConfig({ shouldForwardProp })(
+type StyledFieldProps = {
+  checkableField: boolean
+  flexDirection: flexDirection
+  checked: boolean
+  size: Size
+}
+
+export const Field = styled('div').withConfig({ shouldForwardProp })<StyledFieldProps>(
   props => css`
```
Comment (theo-mesnil) [inline packages/Field/styles.ts:30]:
```suggestion
 ({ flexDirection, checkableField, checked }) => css`
```

File: packages/Field/styles.ts [hunk 5]
```diff
@@ -17,7 +19,14 @@ const checkableFieldStyles = css`
   margin-bottom: sm;
 `
 
-export const Field = styled('div').withConfig({ shouldForwardProp })(
+type StyledFieldProps = {
+  checkableField: boolean
+  flexDirection: flexDirection
```
Comment (theo-mesnil) [inline packages/Field/styles.ts:24]:
```suggestion
flexDirection: WuiProps['flexDirection']
```

File: packages/Field/styles.ts [hunk 6]
```diff
@@ -17,7 +19,14 @@ const checkableFieldStyles = css`
...
+  checked: boolean
+  size: Size
```
Comment (theo-mesnil) [inline packages/Field/styles.ts:26]: All is required ?

File: packages/Field/utils.ts [hunk 7]
```diff
@@ -7,9 +7,21 @@ const TYPES = {
...
 }
 
-const isPristine = (connected, touched) => connected && !touched
+const isPristine = (connected: boolean, touched: boolean) => connected && !touched
 
-export const getBaseType = type => TYPES[type] || type
+export const getBaseType = (type: string): string => TYPES[type] || type
+
+type VariantProps = {
+  connected: boolean
```
Comment (theo-mesnil) [inline packages/Field/utils.ts:15]: all is required ?

File: packages/Field/styles.ts (hunk 6 above)
Comment (cnairiwttj) [reply packages/Field/styles.ts:27]: Oui, toutes les props lui sont passées sauf erreur de ma part

**Review by theo-mesnil — APPROVED** [review]

**Review by mleralec — APPROVED** [review]

File: packages/Field/index.tsx [hunk 8]
```diff
@@ -2,26 +2,25 @@ import React, { forwardRef, Fragment } from 'react'
...
 // Fields
 import { RowContainer } from './layout'
 import * as S from './styles'
 import { getBaseType, getVariant, VariantReturn } from './utils'
 
-export type flexDirection = 'row' | 'container' | 'column'
-
 export type Size = 'sm' | 'md' | 'lg'
 
 export interface FieldProps {
```
Comment (mleralec) [inline packages/Field/index.tsx:14]:
```
export interface FieldOptions {
...
}

export type FieldProps = FieldOptions & React.HTMLAttributes<HTMLInputElement> & WuiProps
```

**Pull request merged**